src/python/WMCore/JobSplitting/EventAwareLumiBased.py

```diff
@@ -309,3 +312,32 @@ def algorithm(self, *args, **kwargs):
         self.lumiChecker.closeJob(self.currentJob)
         self.lumiChecker.fixInputFiles()
         return
+
+    def getFilesSortedByLocation(self, eventsPerJob):
+        """
+        _getFilesSortedByLocation_
+
+        Retrieves a list of files available and sort them by location.
+        If the fileset is closed, resume the splitting. Otherwise check whether
+        there are enough events in each of these locations, if events don't
+        match the desired events_per_job splitting parameter, then skip those
+        files until further cycles.
+        :param eventsPerJob: number of events desired in the splitting
+        :return: a dictionary of files, key'ed by a frozenset location
+        """
+        lDict = self.sortByLocation()
+
+        fileset = self.subscription.getFileset()
```

Comment:
I guess CRAB relies on DataStruct, so I probably have to check whether it's using WMCore.WMBS first (?)